lib/client.js

```diff
@@ -1273,12 +1361,14 @@ function _resume (client, sync) {
       return
     }
 
-    if (!socket) {
+    if (!socket && !client[kHTTP2Session]) {
       connect(client)
       return
     }
 
-    if (socket.destroyed || socket[kWriting] || socket[kReset] || socket[kBlocking]) {
+    if ((socket.destroyed || socket[kWriting] || socket[kReset] || socket[kBlocking]) ||
+      (client[kHTTP2Session] && client[kHTTP2Session].destroyed)) {
+      // TODO(HTTP/2): what if exceeds max concurrent streams or can't accept new
```

Comment:
if possible I'd like to avoid special cases here... socket `socket` be `client[kHTTP2Session]` in case of http2?

Reply:
It can but it touches several places and might lead to more changes, especially as we still handle the socket lifetime (and in userland, a custom socket factory can be provided).
Although destroying the socket will eventually lead to destroying the session as well, so it might not be needed after all 🤔

Let me test it out